src/geckomat/model_adapter/ModelAdapterManager.m

```diff
@@ -0,0 +1,71 @@
+%Abstract Base class for adapters for different species
+classdef ModelAdapterManager 
+	methods(Static)
+        %not really needed, we could access it directly. Not very nice, but practical
+        function adapter = getAdapterFromPath(adapterPath, addToMatlabPath)
+            if nargin < 2
+                addToMatlabPath = true;
+            end
+            
+            potentialAdapterFiles = dir(fullfile(adapterPath,'*.m')); %gets all wav files in struct
```

Comment:
why not be more specific ? e.g. all the adapters should be named as *GEMAdapter.m, so if an user have more in that folder, that files are not added. We then in the repo instructions we mention that.

Reply:
I think that was a criticism @johan-gson had with the earlier implementation, that using the same name actually causes problems?

It is fine if the adapter is set as default and not specified as parameter when running a function, as when you set the default parameter you explicitly refer to a certain path. But if the adapter is specified as function parameter, it should be findable from the MATLAB path, and in that case it should be a unique name. Otherwise having multiple model-specific folders in MATLAB path would cause conflicts.